testcases/alternativesTests/binaries_test_methods.py

```diff
@@ -139,24 +139,18 @@ def all_jre_in_sdk_check(self, args=None):
         for subpkg in jre_subpackages:
             for sdk_subpkg in sdk_subpackages:
                 cont = False
-                for suffix in tc.get_debug_suffixes():
-                    try:
-                        if suffix in subpkg and suffix in sdk_subpkg:
-                            current_subpkg = subpkg
-                            jre = self.installed_binaries[current_subpkg]
-                            current_subpkg = sdk_subpkg
-                            sdk = self.installed_binaries[current_subpkg]
-                            break
-                        elif suffix not in subpkg and suffix not in sdk_subpkg:
-                            current_subpkg = sdk_subpkg
-                            sdk = self.installed_binaries[current_subpkg]
-                            current_subpkg = subpkg
-                            jre = self.installed_binaries[current_subpkg]
-                        else:
-                            cont = True
-                            break
-                    except KeyError:
-                        la.LoggingAccess().log("Subpkg " + current_subpkg + " not containing binaries and is probably "
+                try:
+                    if self._extract_suffix_from_subpkg(subpkg) == self._extract_suffix_from_subpkg(sdk_subpkg):
+                        current_subpkg = subpkg
+                        jre = self.installed_binaries[current_subpkg]
+                        current_subpkg = sdk_subpkg
+                        sdk = self.installed_binaries[current_subpkg]
+                        break
+                    else:
+                        cont = True
+                        break
+                except KeyError:
+                    la.LoggingAccess().log("Subpkg " + current_subpkg + " not containing binaries and is probably "
```

Comment:
here I deal with the bug, where in the case the user tested only a subset, without "debug" suffixed packages in it, the test would fail, as the for cycle wouldn't even iterate once -> jre variable would not be declared.. i removed the inner for cycle and added a support method for extracting the suffix from subpkg (in this particular case, might move it later) and compare only if the suffixes are the same, then proceed